src/media/js/views/search.js

```diff
@@ -117,6 +117,7 @@ define('views/search',
         };
 
         var query = $('#search-q').val() || $('#search-q-desktop').val();
+        var query = query.replace( /  +/g, ' ' );
```

Comment:
This only handles literal spaces, not any whitespace (space, tab, vertical tab, newline, carriage return, formfeed). You maybe should use \s to cover the bases.

I'll defer to @spasovski on whether this functions well enough in the most basic case (' ') to be sufficient.